src/helics/application_api/Endpoints.hpp

```diff
@@ -175,11 +175,19 @@ class Endpoint
     const std::string &getType () const { return fed->getEndpointType (*this); }
     /** get the actual endpoint id for the fed*/
     interface_handle getHandle () const { return handle; }
+    /** implicit conversion operator for extracting the handle*/
+    operator interface_handle () const { return handle; }
```

Comment:
the options were to have it be explicit or have several function calls be duplicated for each interface type.    setInfo,  setOption, getOption.    Otherwise you have an inconsistent interface, and then it gets really ugly with the overloads that are different from Federate, and the messageFederate and ValueFederate, and even more so for the combinationFederate.    We can debate it but I went with what I thought was the lesser of two evils   a implicit conversion operator  or a mess of functions and overloads that would be easy to break if anything changes.

Reply:
That seems fine then. My initial thought was that this should probably be `explicit operator interface_handle ()` and compel the implementer to do their own `static_cast<>()` but that's just me being pedantic.  This is unlikely to cause any real issues as long as it is used safely, so I'm not particularly worried about it, I just wanted to make sure we'd considered possible side effects.